lib/ansible/plugins/action/__init__.py

```diff
@@ -624,11 +629,32 @@ def _fixup_perms2(self, remote_paths, remote_user=None, execute=True):
         # macOS chmod's +a flag takes its own argument. As a slight hack, we
         # pass that argument as the first element of remote_paths. So we end
         # up running `chmod +a [that argument] [file 1] [file 2] ...`
-        res = self._remote_chmod([chmod_acl_mode] + remote_paths, '+a')
+        try:
+            res = self._remote_chmod([chmod_acl_mode] + remote_paths, '+a')
+        except AnsibleAuthenticationFailure as e:
+            # Solaris-based chmod will return 5 when it sees an invalid mode,
+            # and +a is invalid there. Because it returns 5, which is the same
+            # thing sshpass returns on auth failure, our sshpass code will
+            # assume that auth failed. If we don't handle that case here, none
+            # of the other logic below will get run. This is fairly hacky and a
+            # corner case, but probably one that shows up pretty often in
+            # Solaris-based environments (and possibly others).
+            pass
+        else:
```

Comment:
This `else` statement _may_ not be necessary. The difference is subtle, but would behave the same if `rc` is always `!= 0` when an `AnsibleAuthenticationFailure` is raised.

Reply:
The `else` here is mainly meant to be a human reading aid to show that this grouped with the stuff in the `try`. Otherwise as you said, I think it's not super obvious that that logic path depends on results from before this.